syntax = "proto3";

package NSound.NClient;

import "protos/common/directives.proto";
import "protos/common/property-list.proto";
import "protos/common/stream-configuration.proto";

message TStreamMessage {

    message TPush {
        bytes data = 1;
        uint64 size = 2; // in samples
    }

    message TPull {
        uint32 size = 1; // in samples too
    }

    message TConnect {
        NCommon.TStreamConfiguration configuration = 1;
        NCommon.TPropertyList property_list = 2;
    }

    message TClose {

    }

    oneof Request {
        TPush push = 1;
        TPull pull = 2;
        TConnect connect = 3;
        NCommon.TStreamDirective directive = 4;
        TClose close = 5;
    }

    uint32 stream_id = 6;
}